lib/delta-sync/file-sync.js

```diff
@@ -0,0 +1,1051 @@
+import { query, update, sparqlEscapeUri, sparqlEscapeString } from 'mu';
+import { querySudo, updateSudo } from '@lblod/mu-auth-sudo';
+import { escapeRDFTerm, storeError } from "../utils.js";
+import http from "http";
+import fs   from "fs";
+import path from "path";
+import {
+  DISABLE_FILE_INGEST,
+  PREFIXES,
+  TEMP_FILE_GRAPH,
+  SYNC_BASE_URL,
+  DOWNLOAD_FILE_PATH,
+  FILE_FOLDER,
+  INGEST_GRAPH,
+  TEMP_FILE_REMOVAL_GRAPH,
+  REMAPPING,
+  MU_CALL_SCOPE_ID_FILE_SYNC
+} from '../../config.js';
+
+/**
+ * Start the synchronisation of files. Scan the temporary graph, download files and push file metadata to the rest of the application.
+ * Runs different functions to accomplish subtasks.
+ *
+ * @public
+ * @async
+ * @function startFileSync
+ * @return {undefined} - No useful return value, side effects only
+ */
+export async function startFileSync() {
+  console.info(`DISABLE_FILE_INGEST: ${DISABLE_FILE_INGEST}`);
+  try {
+    if (!DISABLE_FILE_INGEST) {
+     await runFileSync(); 
+    }
+    else {
+      console.warn('Automated file ingest disabled');
+    }
+  }
+  catch (err) {
+    console.error("Synchronisation of files failed with error:", err);
+    //Store the error to the database, also include the stacktrace for better debugging (even though the stack trace might be useless when the code changes over time).
+    await storeError(err.message + "\n" + err.stack.toString());
+  }
+}
+
+/**
+ * Run the actual file synchronisation. It downloads files, puts them in the correct location, buffers file metadata until all is received, removes files if metadata is deleted, and also buffers this data.
+ * Calls different functions to accomplish subtasks.
+ *
+ * @private
+ * @async
+ * @function runFileSync
+ */
+async function runFileSync() {
```

Comment:
Error handling is now improved with StackableError. This will (hopefully) give useful error messages about what files failed, but still process them in batch as much as possible for better performance(?) and simplicity.

Reply:
I don't see how StackableError can make it more simple. If you consider the operations on one single file as block in a function, you should have all the info required to elaborate on your error messages -if needed- without breaking the full loop.

Reply:
The StackableError is not breaking the loop. There is an error message per file, if any files failed processing.